src/main/java/com/cloudbees/hudson/plugins/folder/AbstractFolderDescriptor.java

```diff
@@ -216,9 +216,41 @@ public boolean isLookAndFeelConfigurable(AbstractFolder<?> folder) {
      */
     // TODO figure out how one could un-wind name mangling if one ever wanted to
     // TODO move the name mangling code from branch-api to this plugin so that everyone can use it
-    @CheckForNull
+    @NonNull
     public <I extends TopLevelItem> ChildNameGenerator<AbstractFolder<I>,I> childNameGenerator() {
-        return null;
+        return new LegacyChildNameGenerator<>();
```

Comment:
Overridden by `MultiBranchProjectDescriptor` and `OrganizationFolder.DescriptorImpl`, OK.